src/prefect/agent/docker/agent.py

```diff
@@ -130,9 +134,22 @@ def __init__(
             self.host_spec,
         ) = self._parse_volume_spec(volumes or [])
 
-        # Add containers to a docker network
+        # Add containers to the given Docker networks
+        if networks and network:
+            raise ValueError(
+                "Only provide either `network` or `networks` argument, not both!"
+            )
+        if network:
+            warnings.warn(
+                "DockerAgent `network` argument is deprecated and will be removed from Prefect. "
+                "Use `networks` instead.",
+                UserWarning,
+            )
+        # TODO: Only keep `self.networks` when `network` parameter gets removed.
```

Comment:
Not sure if the TODO comment is needed, in the standard deprecation cycle we would probably know what to remove when this comes around 🙂

Reply:
Removed the TODO